src/litserve/server.py

```diff
@@ -153,11 +153,50 @@ def run_streaming_loop(lit_api, request_queue: Queue, request_buffer):
             pipe_s.send((pickle.dumps(e), LitAPIStatus.ERROR))
 
 
+def run_batched_streaming_loop(lit_api, request_queue: Queue, request_buffer, max_batch_size, batch_timeout):
```

Comment:
let's add a comment with a caveat: just like the batched (non-streaming) loop, this will keep generating until all generations in the batch are done

we'll need "continuous batching" to solve this, let's add an issue about it

Reply:
as a more immediate improvement, we need to find a way to avoid sending a lot of tokens past the last token for a particular item in the batch (i.e. we need to trim past the EOS in `encode_response`, let's open an issue and create an example about it)

Reply:
Yeah, I will check on what we can do on here. I have created an issue for now - https://github.com/Lightning-AI/litserve/issues/57